onboard/nav/avoidances.cpp

```diff
@@ -0,0 +1,164 @@
+/* Searches.cpp */
+
+#include "avoidances.hpp"
+#include "utilities.hpp"
+#include "stateMachine.hpp"
+
+#include <iostream>
+#include <cmath>
+#include <stack>
+
+/*************************************************************************/
+/* Avoidance Factory */
+/*************************************************************************/
+// Avoidance Factory to allow for creation of obstacle avoidance objects.
+// Allows for easy change from obstacle avoidance algorithm to another
+Avoidance* AvoidFactory( StateMachine* stateMachine, AvoidanceType type )
+{
+    Avoidance* avoid = nullptr;
+    switch (type)
+    {
+        case AvoidanceType::ORIGINAL:
+            avoid = new Original( stateMachine );
+            break;
+
+        case AvoidanceType::UNKNOWN:
+            std::cerr << "Unkown Search Type. Defaulting to original\n";
+            avoid = new Original( stateMachine );
+            break;
+    }
+    return avoid;
+}
+
+/*************************************************************************/
+/* Original Obstacle Avoidance Algorithm
+   When an obstacle is detected, the rover will drive around whichever way
+   CV decides is the fastest. If the tennisball is detected, the heirarchy has
+   two cases in which the rover will begin to turn to tennis ball:
+        1. the difference in distance (tennis ball is closer than obs)
+        2. Turning around obs is same direction to turn to tennis ball */
+/*************************************************************************/
+Original::~Original() {}
+
+/*****************************************************/
+/* Helpers */
+/*****************************************************/
+
+// Executes the logic for turning around an obstacle.
```

Comment:
If possible, make this comment more concise. It doesn't need to lay out every detail of the function. The assumption does not need to be condensed as it is very important to convey that message.